### ISSUE TITLE
[Dooble] Localization support is broken.

### ISSUE THREAD

**Issue by OscarL** [haikuports/haikuports#9964]
As seen on the [forum](https://discuss.haiku-os.org/t/dooble-browser-localization-can-not-be-fixed-easily/14425), using Dooble in anything but English seems like a mission impossible:

1. Dooble tries to find the `.qm` files under:
   - `/boot/home/Translations` when started via Deskbar/Tracker.
   - under a `Translations` subdir of the current working directory when started from the Terminal.

2. Forcing the search path with: `export DOOBLE_TRANSLATIONS_PATH=/boot/system/apps/Dooble/Translations` fixes part of the issue, but even when `locale -l` returns plain `es`... Dooble wants to find a `dooble_es_ES.qm` file.

   Sounds like [QLocale::system().name()](https://github.com/textbrowser/dooble/blob/7e84b4657c729db80eadf3aef86a74905b1d574d/Source/dooble_settings.cc#L223) is returning `es_ES` even for plain "Español" (same thing with `hu_HU` vs `hu`, etc).

3. After doing:

   ```bash
   extern DOOBLE_TRANSLATIONS_PATH=/tmp/Dooble`
   cp /boot/system/apps/Dooble/Translations/dooble_es.qm /tmp/Dooble/dooble_es_ES.qm
   cp /boot/system/apps/Dooble/Translations/dooble_es.ts /tmp/Dooble/dooble_es_ES.ts
   cp /boot/system/apps/Dooble/Translations/qtbase_es.qm /tmp/Dooble/qtbase_es_ES.qm
   ```

   Dooble gives a: "Warning! the file /tmp/Dooble/dooble_es_ES.qm is perhaps incomplete. The system option has been disabled. English will be assumed. Please read Source/dooble_settings.cc, line 243".

---

Point 1 should be patched on our Doodle `.recipe`, I think.

Point 3 sounds like a Dooble issue, but also... perhaps we shouldn't be packaging translations that can't ever be used?

Point 2... no idea really. Sounds buggy that `QLocale::system::name()` returns `es_ES` for what should be just `es`. Bug on our Qt port? on Haiku's LocaleKit? (in any case... I think Dooble should gracefully pick up `dooble_es.qm` if `dooble_es_ES.qm` can't be found, but it does not has such logic implemented).

Putting everything here for now (for reference, in case we need to open an issue upstream).

**Comment by korli** [comment]
Workaround: in a Terminal, "cd /system/apps/Dooble", then "./dooble". Switch the language to System in Settings, then restart Dooble in the terminal.

**Comment by korli** [comment]
patched in c47ca3997de6c7adf04f61634c1063f8a7f5cd52
